### PR TITLE
Add summarize_pipeline() function and updates tests

### PR DESCRIPTION
**Pull request by ryands02**
Adds a helper function, summarize_pipeline(), to better summarize GPA list, plan to add l1/2/3 models' data/big-picture summaries, have not tested compute environment params because we do not have test data that matches the new format.

Also added minimal_gpa list for testing.
image

### CODE REVIEW

**Comment by ryands02** [comment]
Still need to run build commands to add function export to NAMESPACE and capture documentation.

**Comment by wasciutto** [comment]
We've added the roxygen documentation, so this is ready for your review @michaelhallquist

File: R/summary_funcs.R [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            cat(" -  L2 Models: NA\n")
+          }
+          if (!is.null(gpa$l1_models)) {
+            cat(" -  L3 Models: ", "Present \n")
+          } else {
+            cat(" -  L3 Models: NA\n")
+          }
+          # Computation Info
+          cat("--------\n Computation Info: \n")
+          cat(" FSL Level 1:\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:55]: `gpa$parallel$fsl$l1_feat_time`

File: R/summary_funcs.R [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          cat("  -  Feat Time: ", "\n")
+          cat("  -  Feat Memory (GB): ", "\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:57]: `gpa$parallel$fsl$l1_feat_memgb`

Comment (michaelhallquist) [reply R/summary_funcs.R:57]: Also want `gpa$parallel$fsl$l1_feat_alljobs_time`. This controls the total time the scheduler waits for all L1 jobs to complete before attempting to run L2/L3. The l1_feat_time controls the scheduler time request for each individual L1 job.

File: R/summary_funcs.R [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          cat(" FSL Level 2:\n")
+          cat("  -  Feat Time: ", "\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:59]: `gpa$parallel$fsl$l2_feat_time`

File: R/summary_funcs.R [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          cat("  -  Feat Memory (GB): ", "\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:60]: `gpa$parallel$fsl$l2_feat_memgb`

File: R/summary_funcs.R [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          cat(" FSL Level 3:\n")
+          cat("  -  Feat Time: ", "\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:62]: `gpa$parallel$fsl$l3_feat_time`

File: R/summary_funcs.R [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+          cat("  -  Feat Memory (GB): ", "\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:63]: `gpa$parallel$fsl$l3_feat_memgb`

Comment (michaelhallquist) [reply R/summary_funcs.R:63]: Also want to add `gpa$parallel$fsl$l3_feat_cpusperjob`, which controls how many CPUs are used to run all slices of the L3 model (FLAME breaks the computation into slices along the z axis of the image). The default is 16.

File: R/summary_funcs.R [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+          cat("--------\n System Info: \n")
+          cat(" -  Output Directory: ", gpa$output_directory[1], "\n")
+          cat(" -  System Name: ", gpa$sys_info[1], "\n")
+          cat(" -  Release: ", gpa$sys_info[2], "\n")
+          cat(" -  Version: ", gpa$sys_info[3], "\n")
+          cat(" -  Node Name: ", gpa$sys_info[4], "\n")
+          cat(" -  Machine: ", gpa$sys_info[5], "\n")
+          cat(" -  Login: ", gpa$sys_info[6], "\n")
+          cat(" -  User: ", gpa$sys_info[7], "\n")
+          cat(" -  Effective User: ", gpa$sys_info[8], "\n")
```
Comment (michaelhallquist) [inline R/summary_funcs.R:74]: I would also suggest adding info about the pipeline settings for software, specifically the `$compute_environment` field. This controls the lines added to relevant scripts to setup the environment at the time of compute:

```
$compute_environment
$compute_environment$global
[1] "module use /proj/mnhallqlab/sw/modules"

$compute_environment$r
[1] "module unload r"     "module load r/4.2.1"

$compute_environment$fsl
[1] "module unload fsl"     "module load fsl/6.0.4"

$compute_environment$afni
[1] "module unload afni"       "module load afni/23.0.07"
```

**Review by michaelhallquist — APPROVED** [review]
Great first iteration of this function! I've added some suggestions for additional key fields.

**Pull request merged**